### PR TITLE
DDEV v1.24.0 release announcement

### PR DESCRIPTION
**Pull request by rfay**
## The Issue

DDEV v1.24.0 release announcement

See rendered version at https://20241127-release-v1-24-0.ddev-com-front-end.pages.dev/blog/release-v1240-php84/

Related draft that may need parallel updates, https://github.com/ddev/ddev/issues/6773

## TODO
- [ ] Consider installation instructions

### CODE REVIEW

File: src/content/blog/release-v1.24.0.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+* `ddev describe` changes how exposed ports are displayed for better clarity (thanks @ hanoii)
+* Only a limited set of locales is now installed in ddev-webserver by default. If you need more, use `webimage_extra_packages: [locales-all]`
+* Complex `ddev composer` command usage with json or args with spaces is fixed. (Bug was introduced in v1.23.5)
+
+**Reversions and Removals**
+
+* The `drupal` project type is no longer a generic project type, but instead is an alias to the latest stable Drupal (`drupal11` right now). A number of folks gave feedback that it caused the startup time to be longer and the process more complex. (`ddev config --update` is no longer a part of the Drupal quickstart)
+* In v1.23.5 we started preferring vendor/bin/composer to the specified `composer_version` but users thought that was a mistake, so this was reverted
+* The nginx-proxy router has been removed.
+* The `ddev service enable` and `ddev service disable` commands were removed, as they have long been superceded by the use of `ddev add-on get` and `ddev add-on remove`
```
Comment (tyler36) [inline src/content/blog/release-v1.24.0.md:42]:
```suggestion
* The `ddev service enable` and `ddev service disable` commands were deprecated, as they have long been superceded by the use of `ddev add-on get` and `ddev add-on remove`
```

**Review by tyler36 — CHANGES_REQUESTED** [review]

**Review by stasadev — APPROVED** [review]

File: src/content/blog/release-v1.24.0.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+* `drupal11` is introduced as a project type, demoting the `drupal` project type to be a simple alias to the "latest stable drupal version".
+* The `ddev-webserver` image is at least 25% smaller. Only a few locales are included by default, and only currently-supported versions of PHP are built-in by default (but all the other versions still work). It was 480MB compressed, is now 361MB compressed.
+* `ddev auth ssh` can now be used with individual key files and can follow symlinks. For example, if you only want to have a single file named `id_rsa` to be available to your projects, you can `ddev auth ssh -f ~/.ssh/id_rsa`.
+
+**Smaller Changes**
+
+* Added dynamic timeout handling for scripts on start based on setting of `default_container_timeout`.
+* `ddev describe` changes how exposed ports are displayed for better clarity (thanks @ hanoii)
+* Only a limited set of locales is now installed in ddev-webserver by default. If you need more, use `webimage_extra_packages: [locales-all]`
```
Comment (rpkoller) [inline src/content/blog/release-v1.24.0.md:34]:
```suggestion
* Only a limited set of locales is now installed in `ddev-webserver` by default. If you need more, use `webimage_extra_packages: [locales-all]`
```
line 27 uses single quotes for the image as well

File: src/content/blog/release-v1.24.0.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+* Complex `ddev composer` command usage with json or args with spaces is fixed. (Bug was introduced in v1.23.5)
+
+**Reversions and Removals**
+
+* The `drupal` project type is no longer a generic project type, but instead is an alias to the latest stable Drupal (`drupal11` right now). A number of folks gave feedback that it caused the startup time to be longer and the process more complex. (`ddev config --update` is no longer a part of the Drupal quickstart)
+* In v1.23.5 we started preferring vendor/bin/composer to the specified `composer_version` but users thought that was a mistake, so this was reverted
+* The nginx-proxy router has been removed.
```
Comment (rpkoller) [inline src/content/blog/release-v1.24.0.md:41]:
```suggestion
* The `nginx-proxy` router image has been removed.
```
also wrapped the image in single quotes and also appended image to make it more explicit here.

File: src/content/blog/release-v1.24.0.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
+---
+title: "DDEV Release v1.24.0 with PHP 8.4.1 and all extensions"
+pubDate: 2024-12-02
+# modifiedDate: 2024-07-23
+summary: New features and changes in DDEV v1.24.0
+author: Randy Fay
+featureImage:
+  src: /img/blog/2024/12/php-logo.svg
```
[the remark this answers is not in the document]
Comment (rfay) [reply src/content/blog/release-v1.24.0.md:8]: Thanks, pushed.

File: src/content/blog/release-v1.24.0.md (hunk 3 above)
Comment (rfay) [reply src/content/blog/release-v1.24.0.md:41]: Actually, the image hasn't been removed, the option of using it has been removed.

```suggestion
* The `nginx-proxy` router has been removed and is no longer an option.
```

File: src/content/blog/release-v1.24.0.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+* `drupal11` is introduced as a project type, demoting the `drupal` project type to be a simple alias to the "latest stable Drupal version".
+* The `ddev-webserver` image is at least 25% smaller. Only a few locales are included by default, and only currently-supported versions of PHP are built-in by default (but all the other versions still work). It was 480MB compressed, is now 361MB compressed.
+* `ddev auth ssh` can now be used with individual key files and can follow symlinks. For example, if you only want to have a single file named `id_rsa` to be available to your projects, you can `ddev auth ssh -f ~/.ssh/id_rsa`.
+
+**Smaller Changes**
+
+* Added dynamic timeout handling for scripts on start based on setting of `default_container_timeout`.
+* `ddev describe` changes how exposed ports are displayed for better clarity (thanks [@hanoii](https://github.com/hanoii))
+* Only a limited set of locales is now installed in `ddev-webserver` by default. If you need more, use `webimage_extra_packages: [locales-all]`
```
Comment (ptmkenny) [inline src/content/blog/release-v1.24.0.md:34]: To reduce the image size of `ddev-webserver`, only a few locales are installed by default. To install the full range of locales, set `webimage_extra_packages: [locales-all]` in `.ddev/config.yaml`.

File: src/content/blog/release-v1.24.0.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+* Complex `ddev composer` command usage with json or args with spaces is fixed. (Bug was introduced in v1.23.5)
+
+**Reversions and Removals**
+
+* The `drupal` project type is no longer a generic project type, but instead is an alias to the latest stable Drupal (`drupal11` right now). A number of folks gave feedback that it caused the startup time to be longer and the process more complex. (`ddev config --update` is no longer a part of the Drupal quickstart)
```
Comment (ptmkenny) [inline src/content/blog/release-v1.24.0.md:39]: The `drupal` project type is now an alias to the latest stable major version of Drupal (currently, `drupal11`). A number of folks noted that using `drupal` as a generic project type (the old way `ddev` handled this) caused the startup time to be longer and made the process more complex. (`ddev config --update` is no longer part of the Drupal quickstart.)

**Review by ptmkenny — COMMENTED** [review]
Suggested slightly more detailed explanations for two changes related to Drupal.

**Comment by bmartinez287** [comment]
Screenshot 2024-12-02 at 5 07 59 PM
That PHP pic on top could be better. I like the AI ones a bit more. Plus it being an SVG complicates things since it does not have a background and we rarely used those.

File: src/content/blog/release-v1.24.0.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+* In v1.23.5 we started preferring `vendor/bin/composer` to the specified `composer_version` but users thought that was a mistake, so this was reverted.
+* The nginx-proxy router has been removed.
+* The `ddev service enable` and `ddev service disable` commands were deprecated, as they have long been superceded by the use of `ddev add-on get` and `ddev add-on remove`
+* Removed support for Python and Django4 projects. After more than a year, these did not gain traction or a community, so sadly we had to leave them behind.
+
+Again, thanks so much to all of you for using, promoting, contributing, and supporting DDEV!
```
Comment (bmartinez287) [inline src/content/blog/release-v1.24.0.md:45]: As the DDEV team, we appreciate your support and contributions to DDEV.  Let's work together to improve productivity and performance in 2025!
```suggestion
Again, thanks so much to all of you for using, promoting, contributing, and supporting DDEV!
```

Comment (rfay) [reply src/content/blog/release-v1.24.0.md:45]: I'm desperately trying to figure out what your suggestion was.

Comment (bmartinez287) [reply src/content/blog/release-v1.24.0.md:45]: ups, I just fixed that. I had left it on top.

**Comment by rfay** [comment]
@bmartinez287 I changed it to use the other PHP logo, do you like it better?

**Comment by bmartinez287** [comment]
Yeah!
image
That's the timeless one.

File: src/content/blog/release-v1.24.0.md (hunk 7 above)
Comment (rfay) [reply src/content/blog/release-v1.24.0.md:45]: Sorry, that seems awkward to me. It's longer and less direct. What is it you're trying to accomplish here?

Comment (bmartinez287) [reply src/content/blog/release-v1.24.0.md:45]: Primarily, to reference the DDEV team and skip "so much".

**Comment by rfay** [comment]
@bmartinez287 I went wild and removed twitter/x and added Bluesky. Should have been separate PR but wanted it to be here for this. Could you see if I made any mistakes?

**Comment by bmartinez287** [comment]
Screenshot 2024-12-02 at 7 04 21 PM
We don't have the icon for bluesky

**Comment by rfay** [comment]
If you know how to add one, could you push it in here?

**Comment by rfay** [comment]
It was also missing from any place that wanted it in 
* https://github.com/ddev/ddev/pull/6799

**Pull request merged**